Snail/Scheduler.swift

```diff
@@ -2,24 +2,28 @@
 
 import Foundation
 
-class Scheduler {
+public class Scheduler {
     let delay: TimeInterval
     let repeats: Bool
 
     let event = Observable<Void>()
     private var timer: Timer?
 
-    init(_ delay: TimeInterval, repeats: Bool = true) {
+    public init(_ delay: TimeInterval, repeats: Bool = true) {
         self.delay = delay
         self.repeats = repeats
     }
 
-    @objc private func onNext() {
+    @objc public func onNext() {
         event.on(.next(()))
     }
 
-    func reset() {
+    public func start() {
+        stop()
+        timer = Timer.scheduledTimer(timeInterval: delay, target: self, selector: #selector(onNext), userInfo: nil, repeats: repeats)
```

Comment:
Nice catch on the `repeats` 

![bitmoji](https://render.bitstrips.com/v2/cpanel/67df0485-bbbe-43ca-bc8a-3225873412b3-3a94999d-8a66-4f97-8b45-b4ab35a88bdb-v1.png?transparent=1&palette=1&width=246)